src/common/main.lua

```diff
@@ -1,28 +1,143 @@
 local json = require(".common.json")
 local utils = require(".common.utils")
 local main = {}
+-- just to ignore lint warnings
+local ao = ao or {}
 
 local camel = utils.camelCase
+
+local function errorHandler(err)
+	return debug.traceback(err)
+end
+
 main.init = function()
-	Owner = Owner or ao.env.Process.Owner
-	Name = Name or "ar.io ANT Registry"
+	ANTS = ANTS or {}
+	-- Example ANT structure
+	-- ANTS["antId"] = {
+	--     Owner = "userId",
+	--     Controllers = {"userId1", "userId2"},
+	-- }
+	USERS = USERS or {}
+	-- Example USERS structure
+	-- USERS["userId"] = {"antId1", "antId2"}
 
 	local ActionMap = {
 		Register = "Register",
 		StateNotice = "State-Notice",
 		AccessControlList = "Access-Control-List",
 	}
 
-	Handlers.add("info", Handlers.utils.hasMatchingTag("Action", "Info"), function(msg)
-		ao.send({
-			Target = msg.From,
-			Name = Name,
-			Owner = Owner,
-			Data = json.encode({
-				Name = Name,
-				Owner = Owner,
-			}),
-		})
+	Handlers.add(camel(ActionMap.Register), Handlers.utils.hasMatchingTag("Action", ActionMap.Register), function(msg)
+		print("Action: " .. ActionMap.Register)
+
+		local antId = msg.Tags["Process-Id"]
+		assert(type(antId) == "string", "Process-Id tag is required")
+
+		if ANTS[antId] then
+			ao.send({
```

Comment:
This seems like an overly-stern way to handle this. Why not just lazily create the empty entry if none yet exists and continue with the `State` request and reply with the `Register Notice`? Re-registering seems like a low lift way to update something that might have somehow gotten out of sync.

Reply:
I would agree in other state management scenarios, for example on the users device, but this pattern triggers a "State" message to the ANT - which should be configured to update the ant registry on its own, meaning the state we would get we already have anyways (or we are in the process of retrieving). 

This sternness is to prevent that unnecessary traffic and gas cost - it saves users and the network from abuse.